benchmarks/src/main/scala/caliban/ParserBenchmark.scala

```diff
@@ -1,51 +1,45 @@
 package caliban
 
 import caliban.parsing.Parser
-import cats.effect.IO
-import io.circe.Json
 import org.openjdk.jmh.annotations._
-import sangria.execution._
-import sangria.marshalling.circe._
+import org.openjdk.jmh.infra.Blackhole
 import sangria.parser.QueryParser
 
 import java.util.concurrent.TimeUnit
-import scala.concurrent.duration._
-import scala.concurrent.{ Await, ExecutionContextExecutor, Future }
+import scala.concurrent.ExecutionContextExecutor
 
 @State(Scope.Thread)
 @BenchmarkMode(Array(Mode.Throughput))
 @OutputTimeUnit(TimeUnit.SECONDS)
-@Warmup(iterations = 5, time = 3, timeUnit = TimeUnit.SECONDS)
-@Measurement(iterations = 5, time = 3, timeUnit = TimeUnit.SECONDS)
+@Warmup(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
 @Fork(1)
 class ParserBenchmark {
   import ComplexQueryBenchmark._
 
   implicit val executionContext: ExecutionContextExecutor = scala.concurrent.ExecutionContext.global
 
   @Benchmark
-  def runCaliban(): Unit = {
-    val io = Parser.parseQuery(fullIntrospectionQuery)
-    Caliban.run(io)
+  def runCaliban(bh: Blackhole): Unit = {
+    bh.consume(Parser.parseQueryEither(fullIntrospectionQuery).fold(throw _, identity))
```

Comment:
No need to use an extra blackhole parameter, just returning of results will be enough to use more efficient compiler blackholes, see: https://shipilev.net/jvm/anatomy-quarks/27-compiler-blackholes/

Reply:
Ah nice, thanks for this! I'll change it